### ISSUE TITLE
bar_init width

### ISSUE THREAD

**Issue by ghidra** [lytedev/dotfiles#1]
Hi, 
I have been tinkering with you lemon bar settings for a few days.
One small issue that I ran into was the harcoded geometry size in bar_init line 35.

I dont know bash so much, but I found a script that will use xrandr to get the width, i implemeneted it something like this:

``` bash
#screen width
LINE=`xrandr -q | grep Screen`
WIDTH=`echo ${LINE} | awk '{ print $8 }'`


cat "$BAR_FIFO" | $HOME/.config/bar/bar_format | lemonbar -d -u "$BORDER_WIDTH" -g $(WIDTH)x"$BAR_HEIGHT"+0+0 -f "$BAR_FONT_FAMILY" -f "$BAR_ICON_FONT_FAMILY" -F "$COLOR_FOREGROUND" -B "$COLOR_BACKGROUND" &

```

Though I would let you know incase you found it useful since I noticed your script is quite dynamic itself.

Thanks for sharing.

**Comment by lytedev** [comment]
Good catch! Those darn hardcoded values... sorry about that.

Your exact fix here doesn't work, I think, for those of us with multiple monitors arrange side-by-side, as `xrandr -q | grep Screen | awk '{ print $8 }'` returns `3840`, the width of my two 1920x1080p monitors.

Does this work for you as well? It should return the resolution width of the first monitor xrandr detects as opposed to providing the total resolution. If so, maybe you can create a pull request?

``` bash
xrandr -q | grep " connected " | awk '{ split($2, s, "x"); print s[1] }' | head -n 1
```

**Comment by ghidra** [comment]
quick test just returns "connected". I'll look further into it.

**Comment by lytedev** [comment]
Interesting. Can you post the output of `xrandr -q`?

Here's mine for reference: 

```
Screen 0: minimum 8 x 8, current 3840 x 1080, maximum 16384 x 16384
DVI-I-0 disconnected primary (normal left inverted right x axis y axis)
DVI-I-1 disconnected (normal left inverted right x axis y axis)
DVI-I-2 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 521mm x 293mm
   1920x1080     60.00*+
   1680x1050     59.95  
   1600x900      60.00  
   1440x900      59.89  
   1400x1050     59.98  
   1280x1024     75.02    60.02  
   1280x800      59.81  
   1280x720      60.00  
   1152x864      75.00  
   1024x768      75.03    60.00  
   800x600       75.00    60.32    56.25  
   640x480       75.00    59.94  
HDMI-0 disconnected (normal left inverted right x axis y axis)
DVI-I-3 connected 1920x1080+1920+0 (normal left inverted right x axis y axis) 521mm x 293mm
   1920x1080     60.00*+
   1680x1050     59.95  
   1600x900      60.00  
   1440x900      59.89  
   1400x1050     59.98  
   1280x1024     75.02    60.02  
   1280x800      59.81  
   1280x720      60.00  
   1152x864      75.00  
   1024x768      75.03    60.00  
   800x600       75.00    60.32    56.25  
   640x480       75.00    59.94  
```

**Comment by ghidra** [comment]
basically the same really, just with one monitor:

```
Screen 0: minimum 8 x 8, current 2560 x 1600, maximum 16384 x 16384
DVI-I-0 disconnected (normal left inverted right x axis y axis)
DVI-I-1 connected 2560x1600+0+0 (normal left inverted right x axis y axis) 641mm x 401mm
   2560x1600      60.0*+
   1920x1200      59.9  
   1920x1080      60.0  
   1680x1050      60.0  
   1600x1200      60.0  
   1280x1024      75.0     60.0  
   1280x800       59.8  
   1152x864       75.0  
   1024x768       75.0     60.0  
   800x600        75.0     60.3  
   640x480        75.0     59.9  
DP-0 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 disconnected (normal left inverted right x axis y axis)
DP-3 disconnected (normal left inverted right x axis y axis)
```

**Comment by ghidra** [comment]
ah, this does it:

``` bash
xrandr -q | grep " connected " | awk '{ split($3, s, "x"); print s[1] }' | head -n 1
```

$3 instead of $2

**Comment by ghidra** [comment]
so it appears that this isn't a catch all solution... On my desktop at work, i needed to use $3, and on my laptop i need to use $4, as the line is:
`VGA-0 connected primary 1366x768+0+0 ....`

**Comment by lytedev** [comment]
Resolved with 887a4fb
